File: src/main/java/dev/slne/discord/discord/interaction/modal/step/creator/whitelist/step/WhitelistTicketConfirmTwitchConnected.java
package dev.slne.discord.discord.interaction.modal.step.creator.whitelist.step;

import dev.slne.discord.discord.interaction.modal.step.ModalComponentBuilder;
import dev.slne.discord.discord.interaction.modal.step.ModalSelectionStep;
import dev.slne.discord.message.RawMessages;
import net.dv8tion.jda.api.entities.emoji.Emoji;
import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
import net.dv8tion.jda.api.interactions.components.selections.SelectOption;

public class WhitelistTicketConfirmTwitchConnected extends ModalSelectionStep {

  private static final String TWITCH_CONNECT_TUTORIAL = "https://server.castcrafter.de/support.html#link-twitch";

  private static final String OPTION_YES = "yes";
  private static final String OPTION_NO = "no";

  public WhitelistTicketConfirmTwitchConnected() {
    super(
        RawMessages.get("modal.whitelist.step.twitch.question", TWITCH_CONNECT_TUTORIAL),
        SelectOption.of(
            RawMessages.get("modal.whitelist.step.twitch.question.yes"),
            OPTION_YES
        ).withEmoji(Emoji.fromUnicode("✅")),
        SelectOption.of(
            RawMessages.get("modal.whitelist.step.twitch.question.no"),
            OPTION_NO
        ).withEmoji(Emoji.fromUnicode("❌"))
    );
  }

  @Override
  protected void buildModalComponents(ModalComponentBuilder builder) {
  }

  @Override
  protected void verifyModalInput(ModalInteractionEvent event)
      throws ModalStepInputVerificationException {
    if (getSelected().equals(OPTION_NO)) {
      throw new ModalStepInputVerificationException(
          RawMessages.get("modal.whitelist.step.twitch.error.not-connected"));
    }
  }
}

File: src/main/kotlin/dev/slne/discord/discord/interaction/modal/step/creator/whitelist/step/WhitelistTicketConfirmTwitchConnected.kt
package dev.slne.discord.discord.interaction.modal.step.creator.whitelist.step

import dev.slne.discord.discord.interaction.modal.step.ModalSelectionStep
import dev.slne.discord.exception.step.modal.selection.ValidateModalSelectionException
import dev.slne.discord.message.translatable
import net.dv8tion.jda.api.components.selections.SelectOption
import net.dv8tion.jda.api.entities.emoji.Emoji
import net.dv8tion.jda.api.events.interaction.component.StringSelectInteractionEvent
import net.dv8tion.jda.api.interactions.components.selections.StringSelectInteraction
import net.dv8tion.jda.api.modals.Modal

private const val TWITCH_CONNECT_TUTORIAL = "https://server.castcrafter.de/support.html#link-twitch"
private const val OPTION_YES = "yes"
private const val OPTION_NO = "no"

class WhitelistTicketConfirmTwitchConnected : ModalSelectionStep(
    translatable("modal.whitelist.step.twitch.question", TWITCH_CONNECT_TUTORIAL),
    SelectOption.of(
        translatable("modal.whitelist.step.twitch.question.yes"),
        OPTION_YES
    ).withEmoji(Emoji.fromUnicode("✅")),
    SelectOption.of(
        translatable("modal.whitelist.step.twitch.question.no"),
        OPTION_NO
    ).withEmoji(Emoji.fromUnicode("❌"))
) {
    override fun buildModalComponents(builder: Modal.Builder, interaction: StringSelectInteraction) {
    }

    override suspend fun afterSelection(event: StringSelectInteractionEvent) {
        if (OPTION_YES != selected) {
            throw ValidateModalSelectionException(
                translatable("modal.whitelist.step.twitch.error.not-connected")
            )
        }
    }
}
